modules/rte_mis_school/rte_mis_school.module

```diff
@@ -640,15 +646,20 @@ function rte_mis_school_form_level_education_details_validate(array &$form, Form
       $medium = $educational_detail_entity->get('field_medium')->value;
       $education_level = $educational_detail_entity->get('field_education_level')->value;
 
-      // Check for conflicting education levels.
-      $combination = $board_type . '|' . $category . '|' . $medium;
-      if (isset($education_levels_checked[$combination]) && $education_levels_checked[$combination] == $education_level) {
-        if (in_array($education_level, $education_levels_checked[$combination])) {
-          $form_state->setErrorByName("field_education_details][$key][subform][field_education_level", t('Different entries of Education Level are not allowed for the same Board Type, Education Type, and Medium.'));
-          return;
+      $categories_to_check = ($category === 'co-ed') ? ['boys', 'girls'] : [$category];
+      foreach ($categories_to_check as $cat) {
```

Comment:
I think we should move this to separate function as we are writing same code twice. A function that just checks if this combination already exists or not.

Reply:
ok